### PR TITLE
Not Found Page Added

### PR DESCRIPTION
**Pull request by UgurCanTiryaki**
Not Found page is created to match all the paths that have no corresponding page.

`*` routes are not allowed anymore in vue, therefore `pathMatch` syntax is used.

### CODE REVIEW

**Review by MansurAliKoroglu — CHANGES_REQUESTED** [review]
I see you used `font-mono`.  Are you planning to use that font in whole application or is it just for `NotFound` page specific? Same question for background color.

File: src/components/NotFound.vue [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  <div class="h-screen bg-gray-100 flex flex-col justify-center items-center">
+    <p class="text-center font-mono">
+      <span class="text-9xl">404</span>
+      <br />
+      <span class="text-2xl">Page Not</span>
+      <br />
+      <span class="text-2xl">Found</span>
+      <br />
+      <br />
+      <i class="text-xl">
```
Comment (MansurAliKoroglu) [inline src/components/NotFound.vue:11]: Sonarlint shows me a warning here.

```
Replace this <i> tag by <em>.sonarlint(Web:BoldAndItalicTagsCheck)
```

File: src/components/NotFound.vue [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        <router-link to="/">Click Here to Go to Home</router-link>
+      </i>
+    </p>
```
Comment (MansurAliKoroglu) [inline src/components/NotFound.vue:14]: Usings `<br>` seemed incorrect to me. Maybe you can just use `flex` and `margin`s for styling istead of `<br>`. Also `<span>` inside `<p>` seemed weird to me.

**Review by MansurAliKoroglu — CHANGES_REQUESTED** [review]

**Comment by UgurCanTiryaki** [comment]
> I see you used `font-mono`. Are you planning to use that font in whole application or is it just for `NotFound` page specific? Same question for background color.

The font is page specific. I am not sure about using background as main theme but I think I will use it.

**Comment by MansurAliKoroglu** [comment]
> 
> 
> > I see you used `font-mono`. Are you planning to use that font in whole application or is it just for `NotFound` page specific? Same question for background color.
> 
> The font is page specific. I am not sure about using background color as main theme but I think I will use it.

Well, then you can make `bg-color` global.

File: src/components/NotFound.vue [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<template>
+  <div class="h-screen flex flex-col justify-center items-center">
+    <div class="flex flex-col justify-center items-center mb-10 font-mono">
+      <div class="text-9xl">404</div>
+      <div class="text-3xl">Page Not</div>
+      <div class="text-3xl">Found</div>
+    </div>
+    <router-link to="/" class="text-xl font-mono italic">
```
Comment (MansurAliKoroglu) [inline src/components/NotFound.vue:8]: It's better to use named routes.

File: src/components/NotFound.vue [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<template>
+  <div class="h-screen flex flex-col justify-center items-center">
+    <div class="flex flex-col justify-center items-center mb-10 font-mono">
+      <div class="text-9xl">404</div>
+      <div class="text-3xl">Page Not</div>
+      <div class="text-3xl">Found</div>
```
Comment (MansurAliKoroglu) [inline src/components/NotFound.vue:6]: Is there any reason you are using div over span? Span seems a better approach for simple display things like that.

File: src/components/NotFound.vue (hunk 3 above)
Comment (MansurAliKoroglu) [inline src/components/NotFound.vue:8]: We are applying `font-mono` for whole page so there is no need to put it in two places.

**Review by MansurAliKoroglu — CHANGES_REQUESTED** [review]

File: src/components/NotFound.vue (hunk 4 above)
Comment (UgurCanTiryaki) [reply src/components/NotFound.vue:6]: I want those text one under the other. Span is inline, therefore I thought div is easier to use in this case.

Comment (MansurAliKoroglu) [reply src/components/NotFound.vue:6]: Well, you use flex so end result should be the same. Generally `<div>` is used when you have more elements inside. But for texts like that without no more elements nested, div seems a bit redundant.

**Review by MansurAliKoroglu — APPROVED** [review]

**Pull request merged**